ADR/0006-log-conventions.md

```diff
@@ -11,39 +12,36 @@ Accepted
 
 We need log conventions to make our offering easier to operate and maintain. These will also make it possible for us to create queries that cross components in our logs, which supports [STONE-201](https://issues.redhat.com/browse/STONE-201).
 
-## Decision: Log conventions
+We initially decided to require log messages formatted by the `console` encoder, thinking that it would be advantageous to be able to consume logs in a human readable format directly from our controller pods (2023-03-08), but on [subsequent review](https://redhat-internal.slack.com/archives/C02C3SE8QS2/p1680074942662149?thread_ts=1680071345.179399&cid=C02C3SE8QS2) we realized that this bears an undesirable cost in configuration of our logging: the default encoder for the "production" profile in zap is the `json` encoder (2023-03-29).
 
-In our controller logs, we will use structured log messages, partially formatted in JSON with key-value pairs as described below.
+## Decision: Log conventions
 
-```
-TIMESTAMP LOG_LEVEL LOGGER MESSAGE ADDITIONAL_JSON_DATA
-```
+In our controller logs, we will use structured log messages, formatted in JSON with key-value pairs as described below.
 
 For example:
 
 ```
-2023-03-07T11:32:29.167Z    INFO    controllers.Component   updating devfile component name kubernetes-deploy ...   {"namespace": "samburrai-tenant", "resource": "go-net-http-hello-prqx", "kind": "Component"}
-2023-03-07T11:32:29.020Z    INFO    controllers.Component   API Resource updated {"namespace": "samburrai-tenant", "audit": "true", "resource": "go-net-http-hello-prqx", "kind": "Component", "action": "UPDATE"}
+{"ts": "2023-03-07T11:32:29.167Z", "level": "info", "logger": "controllers.Component", "msg": "updating devfile component name kubernetes-deploy ...", "namespace": "samburrai-tenant", "name": "go-net-http-hello-prqx", "controllerKind": "Component"}
+{"ts": "2023-03-07T11:32:29.020Z", "level": "info", "logger": "controllers.Component", "msg": "API Resource updated", "namespace": "samburrai-tenant", "audit": "true", "name": "go-net-http-hello-prqx", "controllerKind": "Component", "action": "UPDATE"}
 ```
 
-The meaning of **timestamp** and log level should be self evident. The **logger** name helps your team understand which part of your controller is emitting the log (usually `Log.WithName` from `zap`). The **message** is a human readable string describing the event being logged. The **json data** contains additional fields useful for searching.
+The **ts** field stands for "timestamp" (encoded as UTC/ISO-8601) and the meaning of log **level** should be self evident. The **logger** name helps your team understand which part of your service is emitting the log (usually `Log.WithName` from `zap`). The **msg** is a human readable string describing the event being logged. Further **json** key value pairs contain additional fields useful for searching.
```

Comment:
Only a part of allowed fields are described here, which is confusing to me. Should we have all possible fields explained in one pace, so it's easy to find what is what?

Reply:
That can be found down in the `# Appendix`.

Reply:
Can we add something like "See full list of fields in Appendix"?

Reply:
Yes, good idea. Will do.